contracts/orderbook/Logic/OrderBook.sol

```diff
@@ -0,0 +1,413 @@
+pragma solidity ^0.8.0;
+import "../Storage/OrderBookEvents.sol";
+import "../Storage/OrderBookStorage.sol";
+import "../../swaps/ISwapsImpl.sol";
+import "../OrderVault/IDeposits.sol";
+import "../../interfaces/IDexRecords.sol";
+import "../../mixins/Flags.sol";
+contract OrderBook is OrderBookEvents, OrderBookStorage, Flags {
+    using EnumerableSet for EnumerableSet.Bytes32Set;
+
+    function initialize(address target) public onlyOwner {
+        _setTarget(this.getSwapAddress.selector, target);
+        _setTarget(this.getFeed.selector, target);
+        _setTarget(this.getDexRate.selector, target);
+        _setTarget(this.clearOrder.selector, target);
+        _setTarget(this.prelimCheck.selector, target);
+        _setTarget(this.queryRateReturn.selector, target);
+        _setTarget(this.priceCheck.selector, target);
+        _setTarget(this.executeOrder.selector, target);
+        _setTarget(this.setVaultAddress.selector, target);
+    }
+
+    function _executeTradeOpen(
+        IOrderBook.Order memory internalOrder
+    ) internal {
+        IDeposits(vault).withdraw(internalOrder.orderID);
+        (bool result, bytes memory data) = internalOrder.iToken.call(
+            abi.encodeWithSelector(
+                IToken(internalOrder.iToken).marginTrade.selector,
```

Comment:
`internalOrder` comes externally. make sure there is no possibility of reentrancy

Reply:
in the checks when placing orders it checks that the iToken contract specified is a loan pool